src/index.js

```diff
@@ -1,122 +1,157 @@
 
-// The start function is run by the BaseKonnector instance only when it got all the account
-// information (fields). When you run this connector yourself in "standalone" mode or "dev" mode,
-// the account information come from ./konnector-dev-config.json file
+const requestPromise = require('request-promise')
+const cookiejar = requestPromise.jar()
+
+const request = (() => {
+  let rp
+  function init() {
+    requestPromise.defaults({ jar: cookiejar })
+    return requestPromise
+  }
+  return (() => {
+    if (!rp) {
+      rp = init()
+    }
+    return rp
+  })()
+})()
+
 async function start(fields) {
   log('info', 'Authenticating ...')
   await authenticate(fields.login, fields.password)
   log('info', 'Successfully logged in')
-  // The BaseKonnector instance expects a Promise as return of the function
   log('info', 'Fetching the list of documents')
-  const $ = await request(`${baseUrl}/index.html`)
-  // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
+  const $ = await request({
+    uri: `${baseURL.origin}/fr/orders`,
+    transform: body => cheerio.load(body),
+    jar: cookiejar
+  })
   log('info', 'Parsing list of documents')
   const documents = await parseDocuments($)
 
-  // here we use the saveBills function even if what we fetch are not bills, but this is the most
-  // common case in connectors
   log('info', 'Saving data to Cozy')
-  await saveBills(documents, fields.folderPath, {
-    // this is a bank identifier which will be used to link bills to bank operations. These
-    // identifiers should be at least a word found in the title of a bank operation related to this
-    // bill. It is not case sensitive.
-    identifiers: ['books']
-  })
-}
-
-// this shows authentication using the [signin function](https://github.com/konnectors/libs/blob/master/packages/cozy-konnector-libs/docs/api.md#module_signin)
-// even if this in another domain here, but it works as an example
-function authenticate(username, password) {
-  return signin({
-    url: `http://quotes.toscrape.com/login`,
-    formSelector: 'form',
-    formData: { username, password },
-    // the validate function will check if
-    validate: (statusCode, $) => {
-      // The login in toscrape.com always works excepted when no password is set
-      if ($(`a[href='/logout']`).length === 1) {
-        return true
-      } else {
-        // cozy-konnector-libs has its own logging function which format these logs with colors in
-        // standalone and dev mode and as JSON in production mode
-        log('error', $('.error').text())
-        return false
+  await saveBills(
+    documents.map(entry => ({
+      ...entry,
+      requestOptions: {
+        jar: cookiejar
       }
+    })),
+    fields.folderPath,
+    {
+      identifiers: ['deliveroo']
     }
+  )
+}
+
+async function authenticate(username, password) {
+  const options = {
+    formselector: 'form#login',
+    input: {
+      login: 'login_email',
```

Comment:
This is weird : login: 'login_email'

Reply:
It means that the login input is named `login_email`.
According to the classic form of a web signin form: login / password.

Reply:
I named the configuration object for input fields:
```js
const input = {
  input_login_name: "login_email",
  input_password_name: "login_password"
}
```

Do you find it less weird?